graphics/opengl/context.cpp

```diff
@@ -84,56 +84,77 @@ void Context::initialize(ContextType contextType) {
 	type = contextType;
 
 #ifdef USE_GLAD
+	int gladVersion;
 	switch (type) {
 	case kContextGL:
-		gladLoadGL(loadFunc);
+		gladVersion = gladLoadGL(loadFunc);
 		break;
 
 	case kContextGLES:
-		gladLoadGLES1(loadFunc);
+		gladVersion = gladLoadGLES1(loadFunc);
 		break;
 
 	case kContextGLES2:
-		gladLoadGLES2(loadFunc);
+		gladVersion = gladLoadGLES2(loadFunc);
 		break;
 
 	default:
+		gladVersion = 0;
 		break;
 	}
-#endif
 
-	const char *verString = (const char *)glGetString(GL_VERSION);
+	majorVersion = GLAD_VERSION_MAJOR(gladVersion);
+	minorVersion = GLAD_VERSION_MINOR(gladVersion);
 
-	if (!verString) {
-		majorVersion = minorVersion = 0;
-		warning("Could not parse fetch GL_VERSION: %d", glGetError());
-	} else if (type == kContextGL) {
-		// OpenGL version number is either of the form major.minor or major.minor.release,
-		// where the numbers all have one or more digits
-		if (sscanf(verString, "%d.%d", &majorVersion, &minorVersion) != 2) {
-			majorVersion = minorVersion = 0;
-			warning("Could not parse GL version '%s'", verString);
+	if (!gladVersion)
+		// If gladVersion is 0 it either means:
+		// - loading failed and glad didn't even set up core functions
```

Comment:
This case should be fatal, right? I believe this is addressed by the `glGetString` check below, correct?

Reply:
Indeed. I added the fallback only because of https://github.com/scummvm/scummvm/pull/5512.
In a perfect world, we could error out directly.